Nesting of block indented expressions

topecongiro opened the pull request:
Closes #1422, closes #1449, and closes #1494.

nrc commented on src/expr.rs:
```diff
@@ -1318,18 +1318,21 @@ impl Rewrite for ast::Arm {
             }
-            _ => &**body,
+            ast::ExprKind::Call(_, ref args) => (args.len() == 1, &**body),
+            ast::ExprKind::Closure(..) => (true, &**body),
```
Should we handle tuples here too?

topecongiro replied:
Tuples are not refered in https://github.com/rust-lang-nursery/fmt-rfcs/issues/61, but if we should, do we use "a single expression parameter" rule or allow exception like clousres?
```rust
// A single expression tuple
let a = foo(bar((|x| {
    let y = x + 1;
    let z = y + 1;
    z
})));

// Should we allow this?
let a = foo(bar((b, c, |x| {
    let y = x + 1;
    let z = y + 1;
    z
})));
```

nrc replied:
I think we should allow both (note that tuples with a single element must always have a trailing comma).

topecongiro replied:
Thanks for clarifying!

topecongiro replied:
I would like to make it clear, should we use block indent for tuples when `fn_call_style = Block` and format like function call? (c.f. https://github.com/rust-lang-nursery/fmt-rfcs/issues/64)
```rust
let a = (
    xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
    yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy,
);
```

nrc replied:
yes, I think so

topecongiro commented:
I have added tuple and block. Also added a commit to add a trailing comma to a single arg in multiline.
Now rewriting tuple uses the same function as rewriting function call when `fn_call_style = Block`.

nrc commented:
Awesome, thank you!
